src/TextInput/TextInput.tsx

```diff
@@ -219,6 +219,7 @@ export class TextInput extends React.Component<Props, State> {
         </div>
         <input
           {...additionalProps}
+          id={this.props.name}
```

Comment:
> My one concern is if the TextInput component is currently being used with an id prop passed in as an additional prop, as this would override it.

Great callout! I definitely think that, in an ideal world, this component should always be accessible. And one way to accomplish that is to always set `id` to `this.props.name`.

At the same time, I think that engineers are pretty used to being able to specify `id` for any component and having that stick. And in fact, we have some end-to-end tests that look for elements with specific IDs, so I do still think that we should use the `id` when specified.

Since we ultimately just need the label's `for` attribute to match the input's `id` attribute, another option might be to change what we set the label `htmlFor` prop to, e.g.

```
<label htmlFor={this.props.id || this.props.name}>
<input id={this.props.id || this.props.name}>
```

I think that this works for all cases, e.g. `id` and no `name`, `name` and no `id`, and both.

It's also worth adding a comment to the code noting the need for these to match just to avoid someone accidentally undoing this.

Reply:
Great suggestion, thanks Arsalan!